### PR TITLE
Hide onboarding users

### PR DESCRIPTION
**Pull request by xsyann**

### CODE REVIEW

File: core/server/users.js [hunk 1]
```diff
@@ -9,7 +9,8 @@ const mainFields = {
 }
 
 const getUsers = (levelId) => {
-    let filters = { 'status.online': true, 'profile.levelId': levelId }
+    const registeredOrGuest = {"$or": [{"profile.guest": {"$ne": true}, "username": null}, {"profile.guest": true, "username": {$exists: true}} ]}
+    const filters = { 'status.online': true, 'profile.levelId': levelId, ...registeredOrGuest}
```
Comment (ramnes) [inline core/server/users.js:13]: Why not just something like this?

```suggestion
    const filters = { 'status.online': true, 'profile.levelId': levelId, 'profile.guest': false }
```

Comment (xsyann) [reply core/server/users.js:13]: I didn't want to break the guest feature

Comment (ramnes) [reply core/server/users.js:13]: Also, what is `username`? In production, we have only one user with this field:

```
$ db.users.countDocuments({"username": {$exists: true}})
1
```

Looks like it's a deprecated concept we want to remove from the codebase, isn't it?

Comment (xsyann) [reply core/server/users.js:13]: I don't know if it is used for another purpose but it is used at least to differentiate a guest and a user before on-boarding. `username` is a random name assigned to a guest and then removed when the guest sign-in/up.

Comment (ramnes) [reply core/server/users.js:13]: > I didn't want to break the guest feature

Hm, fair enough, let's keep it that way for now and remove it later if we want to.

**Review by ramnes — APPROVED** [review]

**Pull request merged**